### PR TITLE
Use __make_integer_seq intrinsic when available

### PR DESCRIPTION
**Pull request by CaseyCarter**

### CODE REVIEW

**Comment by ericniebler** [comment]
Merge conflict. :-(

File: include/meta/meta.hpp [hunk 1]
```diff
@@ -3162,12 +3173,15 @@ namespace meta
         /// \f$ O(log(N)) \f$.
         /// \ingroup integral
         template <typename T, T N>
-        using make_integer_sequence =
+        using make_integer_sequence = _t<
             _t<detail::coerce_indices_<T, 0, make_index_sequence<static_cast<std::size_t>(N)>>>;
```
Comment (ericniebler) [inline include/meta/meta.hpp]: `_t<_t<` looks wrong to me. Is this intended?

Comment (CaseyCarter) [inline include/meta/meta.hpp]: It's an advanced technique, you're probably not familiar with it.

Comment (CaseyCarter) [inline include/meta/meta.hpp]: I've found that compile times are greatly improved by deliberately making the program ill-formed. Just think how fast meta will be on Metabench.

**Comment by ericniebler** [comment]
This is very cool, BTW. 👍

File: include/meta/meta.hpp (hunk 1 above)
Comment (gabyx) [inline include/meta/meta.hpp]: I am interested about the additional `_t<` which does that not compile (@CaseyCarter)? Is somebody kind to explain what this is meant for? =) What is the advanced technique here? :-)
BR

Comment (CaseyCarter) [inline include/meta/meta.hpp]: > What is the advanced technique here?

It's just a typo. I was trying - and apparently failing - to be funny.

Comment (gabyx) [inline include/meta/meta.hpp]: lol, the problem, is the content in meta.hpp is sometimes so advanced shi**,  I always stumble across stuff where I think "wtf, whats happening here :-)" which are not at all easy to understand but cool, I've seen this `t_<` -> advanced technique? hää? :-)  thx for the headache relieve :-)

Comment (ldionne) [inline include/meta/meta.hpp]: > It's just a typo. I was trying - and apparently failing - to be funny.

FWIW, I laughed.

Comment (gnzlbg) [inline include/meta/meta.hpp]: > apparently failing - to be funny.

I laughed as well. 

Meta did actually pretty good in the comparison that was posted on reddit the other day, are there metabench results for this yet?

Comment (ldionne) [inline include/meta/meta.hpp]: @gnzlbg Can you please provide a link to the comparison? I've been without internet for a while now so I haven't seen it.

Comment (brunocodutra) [inline include/meta/meta.hpp]: @ldionne I suppose @gnzlbg means [this post](https://www.reddit.com/r/cpp/comments/4k57tk/boost_mpl_dead/) where I mention metabench in one of my replies.

To answer the question, metabench results are updated daily, however there are no benchmarks as of yet that specifically compare the performance of generating sequences.

**Pull request merged**